File: src/main/java/org/example/kalkulationsprogramm/domain/OooReplyLog.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "ooo_reply_log",
        uniqueConstraints = @UniqueConstraint(
                name = "uk_ooo_reply_log_schedule_sender",
                columnNames = { "schedule_id", "sender_address" }))
public class OooReplyLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "schedule_id", nullable = false)
    private Long scheduleId;

    @Column(name = "sender_address", nullable = false, length = 320)
    private String senderAddress;

    @Column(name = "replied_at", nullable = false)
    private LocalDateTime repliedAt;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/OooReplyLog.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table
open class OooReplyLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(name = "schedule_id", nullable = false)
    open var scheduleId: Long? = null

    @Column(name = "sender_address", nullable = false, length = 320)
    open var senderAddress: String? = null

    @Column(name = "replied_at", nullable = false)
    open var repliedAt: LocalDateTime? = null

}